docs/src/apireference.md

```diff
@@ -291,6 +283,39 @@ objective_expr
 constraint_expr
 ```
 
+## Errors
+
+When an MOI call fails on a model, precise errors should be thrown when possible
+instead of simply calling `error` with a message. This allows for instance to
+the CachingOptimizer to automatically detach the optimizer when an
```

Comment:
Mentioning `CachingOptimizer` might be more confusing than helpful in this context. You could say more simply that this error-reporting system allows code to distinguish between internal errors (that should be shown to the user) and unsupported operations which may have automatic workarounds.